### PR TITLE
fix: fix tcp connection may crash before wait

### PR DESCRIPTION
**Pull request by AbelHu**
fix tcp connection may crash before wait

### CODE REVIEW

File: parts/windows/kuberneteswindowsfunctions.ps1 [hunk 1]
```diff
@@ -322,15 +322,19 @@ function Check-APIServerConnectivity {
...
-        if ($tcpClient.Connected) {
-            Write-Log "Retry $retryCount : Connected to API server successfully"
-            return
+        try {
+            $tcpClient=New-Object Net.Sockets.TcpClient
+            Write-Log "Retry $retryCount : Trying to connect to API server $MasterIP"
+            $tcpClient.ConnectAsync($MasterIP, 443).wait($ConnectTimeout*1000)
+            if ($tcpClient.Connected) {
+                Write-Log "Retry $retryCount : Connected to API server successfully"
+                return
```
Comment (mainred) [inline parts/windows/kuberneteswindowsfunctions.ps1:331]: Do we need to close the connection?

**Review by mainred — APPROVED** [review]
left a nit.

File: parts/windows/kuberneteswindowsfunctions.ps1 [hunk 2]
```diff
@@ -322,15 +322,19 @@ function Check-APIServerConnectivity {
...
-        Write-Log "Retry $retryCount : Trying to connect to API server $MasterIP"
-        $tcpClient.ConnectAsync($MasterIP, 443).wait($ConnectTimeout*1000)
-        if ($tcpClient.Connected) {
-            Write-Log "Retry $retryCount : Connected to API server successfully"
-            return
+        try {
+            $tcpClient=New-Object Net.Sockets.TcpClient
+            Write-Log "Retry $retryCount : Trying to connect to API server $MasterIP"
+            $tcpClient.ConnectAsync($MasterIP, 443).wait($ConnectTimeout*1000)
+            if ($tcpClient.Connected) {
```
Comment (mainred) [inline parts/windows/kuberneteswindowsfunctions.ps1:329]: Can we print the tcpClient status when it failed?

Comment (mainred) [reply parts/windows/kuberneteswindowsfunctions.ps1:329]: like connection reset or timedout, etc.

**Review by feiskyer — APPROVED** [review]

File: parts/windows/kuberneteswindowsfunctions.ps1 (hunk 1 above)
Comment (AbelHu) [reply parts/windows/kuberneteswindowsfunctions.ps1:332]: Good point. I think that the connection will be closed when the function exits. Adding close does not harm. added.

File: parts/windows/kuberneteswindowsfunctions.ps1 (hunk 2 above)
Comment (AbelHu) [reply parts/windows/kuberneteswindowsfunctions.ps1:329]: What info do you want to get? I have printed the exception if it crashes. I think that we cannot get more info when connection fails.

**Pull request merged**